Xcode Warnings on Xcode 13.3

### Describe the bug

> 'self' refers to the method 'LeafKeyword.self', which may be unexpected

```
SourcePackages/checkouts/leaf-kit/Sources/LeafKit/LeafLexer/LeafParameterTypes.swift:67:44: warning: 'self' refers to the method 'LeafKeyword.self', which may be unexpected
    internal var identity: Bool { self == .`self` }
                                           ^
SourcePackages/checkouts/leaf-kit/Sources/LeafKit/LeafLexer/LeafParameterTypes.swift:67:44: note: use 'LeafKeyword.self' to silence this warning
    internal var identity: Bool { self == .`self` }
                                           ^
                                           LeafKeyword.
<unknown>:0: warning: 'self' refers to the method 'LeafKeyword.self', which may be unexpected
<unknown>:0: note: use 'LeafKeyword.self' to silence this warning
```

### To Reproduce

Build a Vapor project with `leaf-kit` in it on Xcode 13.3

### Expected behavior

No Xcode warnings should exist

### Environment

vapor: stable 18.3.3

FYI: the issue template says to use `vapor --version` but this command doesn't exist any longer.

--- a/Sources/LeafKit/LeafLexer/LeafParameterTypes.swift
+++ b/Sources/LeafKit/LeafLexer/LeafParameterTypes.swift
@@ -47,15 +47,15 @@ public enum Parameter: Equatable, CustomStringConvertible {
 public enum LeafKeyword: String, Equatable {
     // MARK: Public - Cases
     
-    //               Eval -> Bool / Other
-    //            -----------------------
-    case `in`,    //
-         `true`,  //   X       T
-         `false`, //   X       F
-         `self`,  //   X             X
-         `nil`,   //   X       F     X
-         `yes`,   //   X       T
-         `no`     //   X       F
+    //                      Eval -> Bool / Other
+    //                   -----------------------
+    case `in`,           //
+         `true`,         //   X       T
+         `false`,        //   X       F
+         this = "self",  //   X             X
+         `nil`,          //   X       F     X
+         `yes`,          //   X       T
+         `no`            //   X       F
 
     // MARK: Internal Only
     
@@ -64,12 +64,17 @@ public enum LeafKeyword: String, Equatable {
     internal var isBooleanValued: Bool { [.true, .false, .nil, .yes, .no].contains(self) }
     // Value or value-indicating returns
     internal var `nil`: Bool { self == .nil }
-    internal var identity: Bool { self == .`self` }
+    internal var identity: Bool { self == .this }
     internal var bool: Bool? {
         guard isBooleanValued else { return nil }
         return [.true, .yes].contains(self)
     }
 }
+
+extension LeafKeyword {
+     @available(*, deprecated, message: "Use .this instead")
+     static var `self`: Self { this }
+ }
 
 // MARK: - Operator Symbols
 

--- a/Sources/LeafKit/LeafSerialize/ParameterResolver.swift
+++ b/Sources/LeafKit/LeafSerialize/ParameterResolver.swift
@@ -67,7 +67,7 @@ internal struct ParameterResolver {
                 return data[keyPath: v] ?? .trueNil
             case .keyword(let k):
                 switch k {
-                    case .self: return .init(.dictionary(data))
+                    case .this: return .init(.dictionary(data))
                     case .nil: return .trueNil
                     case .true, .yes: return .init(.bool(true))
                     case .false, .no: return .init(.bool(false))

